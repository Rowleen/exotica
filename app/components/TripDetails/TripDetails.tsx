import { FC } from 'react';
import { Trip } from '../../core/domain/entities/Trip';
import { Itinerary } from '../../core/domain/entities/Itinerary';

import styles from './tripDetails.module.sass';

interface TripDetailsProps {
  trip: Trip;
}

const TripDetails: FC<TripDetailsProps> = ({ trip }) => {
  return (
    <article className={styles.wrapper}>
      <img className={styles.photo} src={trip.photo_url} alt="Trip photo" />

      <div className={styles.info}>
        <h1 className={styles.title}>{trip.title}</h1>

        <div className={styles.action}></div>

        <p className={styles.description}>{trip.description}</p>

        <h2 className={styles.itineraryTitle}>Itinerary</h2>

        {trip.itinerary.map((step: Itinerary, index) => (
          <div key={index + step.day} className={styles.step}>
            <h3 className={styles.stepTitle}>
              Day {step.day}: {step.location}
            </h3>
            <p className={styles.stepDescription}>{step.description}</p>
          </div>
        ))}
      </div>
    </article>
  );
};

export default TripDetails;
