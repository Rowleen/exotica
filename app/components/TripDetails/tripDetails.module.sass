@import '../../sass/variables'

.wrapper
  width: 100%
  display: flex
  flex-direction: column

  .photo
    width: 100%
    height: auto

  .info
    padding: $spacing-4xl $spacing-3xl

    .title
      font-family: 'Roboto Bold'
      font-size: 2rem
      margin-bottom: $spacing-md

    .description
      display: block
      color: $font-color-primary
      line-height: 24px
      border-bottom: solid 1px $border-color-primary
      padding-bottom: $spacing-3xl
      margin-bottom: $spacing-3xl

    .itineraryTitle
      font-size: 1.5rem
      margin-bottom: $spacing-3xl

    .step
      padding-left: $spacing-3xl
      padding-bottom: $spacing-3xl

      .stepTitle, .stepDescription
        font-family: 'Roboto Regular'

      .stepTitle
        color: $font-color-primary
        font-size: 1rem
        font-weight: 400
        margin-bottom: $spacing-md

      .stepDescription
        color: $font-color-disabled
        line-height: 24px
