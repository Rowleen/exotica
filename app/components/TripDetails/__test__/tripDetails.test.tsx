import '@testing-library/jest-dom'
import { render } from '@testing-library/react'
import TripDetails from '../TripDetails'

import tripMock from '../../../__mocks__/trip.mock.json'
import tripWithNoItineraryMock from '../../../__mocks__/tripWithNoItinerary.mock.json'

describe('Home', () => {
  const renderComponent = <TripDetails trip={tripMock} />

  it('renders trip details modal unchanged', () => {
    const { container } = render(renderComponent)
    expect(container).toMatchSnapshot()
  })

  it('render country name', () => {
    const { getByRole } = render(<TripDetails trip={tripMock} />)

    const heading = getByRole('heading', { level: 1 })

    expect(heading).toBeInTheDocument()
  })

  it('render trip description', () => {
    const { getByTestId } = render(<TripDetails trip={tripMock} />)

    const description = getByTestId('description')

    expect(description).toBeInTheDocument()
  })

  it('render itineray days', () => {
    const { getByRole } = render(<TripDetails trip={tripMock} />)

    const itineraryHeading = getByRole('heading', { level: 2 })

    expect(itineraryHeading).toBeInTheDocument()
  })

  it('show message of no itinerary for this trip', () => {
    const { getByTestId, queryByRole } = render(
      <TripDetails trip={tripWithNoItineraryMock} />
    )

    const itineraryHeading = queryByRole('heading', { level: 2 })
    const message = getByTestId('no-itinerary')

    expect(itineraryHeading).not.toBeInTheDocument()
    expect(message).toBeInTheDocument()
  })
})
