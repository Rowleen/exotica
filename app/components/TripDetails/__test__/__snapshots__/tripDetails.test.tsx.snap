// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home renders trip details modal unchanged 1`] = `
<div>
  <article
    class="wrapper"
  >
    <img
      alt="Trip photo"
      class="photo"
      src="https://a.cdn-hotels.com/gdcs/production82/d1923/447a348f-f875-4885-b00a-e9a90603fef5.jpg"
    />
    <div
      class="info"
    >
      <h1
        class="title"
      >
        Portugal
      </h1>
      <div
        class="action"
      />
      <p
        class="description"
        data-testid="description"
      >
        Embark on a journey through Portugal, where the charming streets of Lisbon captivate you, the golden beaches of the Algarve await, and Portuguese cuisine delights with authentic flavors. Explore castles in Sintra and create unforgettable memories in this destination full of history and beauty. Portugal invites you to experience something truly unique!
      </p>
      <div
        class="itineraryWrapper"
      >
        <h2
          class="itineraryTitle"
        >
          Itinerary
        </h2>
        <div
          class="step"
        >
          <h3
            class="stepTitle"
          >
            Day 
            1
            : 
            Lisbon
          </h3>
          <p
            class="stepDescription"
          >
            Explore the Alfama neighborhood and visit São Jorge Castle.
          </p>
        </div>
        <div
          class="step"
        >
          <h3
            class="stepTitle"
          >
            Day 
            2
            : 
            Lisbon
          </h3>
          <p
            class="stepDescription"
          >
            Visit the Jerónimos Monastery and the Monument to the Discoveries.
          </p>
        </div>
      </div>
    </div>
  </article>
</div>
`;
